src/www/system_gateways_edit.php

```diff
@@ -37,9 +37,9 @@
     $a_gateways_arr[] = $gw;
```

Comment:
you can drop this whole file from the PR

Reply:
What ,,, so it becomes

$a_gateways_arr = array();
$a_gateways = return_gateways_array(true, false, true);
if (!isset($config['system']['prefer_dpinger'])) {
...

Reply:
it's ok as is now: return_apinger_defaults is a cheap call and is fixed up when prefer_dpinger is set below

the diff later also rewrites part of unrelated fixes on master